### PR TITLE
Update Kubeapps governance and proposal request process

### PR DESCRIPTION
**Pull request by ppbaena**
### Description of the change

Update Kubeapps governance process to include the proposal process to define how to build consensus on a problem statement and solution design before starting work on the implementation.

### Benefits

Unified process and defined lifecycle for new proposals that could impact on Kubeapps. Make the decision process open and transparent for community users and track record of the engineering decisions within the same github repo, so all information is self-contained.

### Possible drawbacks

- Process needs to be followed by maintainers team.
- New template for proposals increase diversity of issues to be opened, so could cause some confusion to users.

### Applicable issues

- related to #4734 

### Additional information
A new board for proposals will be created to track all issues marked with label `kind/proposal`.

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -6,50 +6,27 @@ This document defines the project governance for Kubeapps.
 
 **Kubeapps**, an open-source project, is committed to building an open, inclusive, productive and self-governing open source community involved to simplifying how applications are deployed and managed in Kubernetes clusters. The community is governed by this document to define how the community should work together to achieve this goal.
 
-## Code Repositories
+## Code of Conduct
```
Comment (antgamdia) [inline GOVERNANCE.md:9]: Some minor typos:

Not here, but above:

`involved TO simplifying`  -> `involved in simplifying`

And at the "credits" section:

`Sections of this documentS have .... ` --> `Sections of this document have .... `

File: site/content/docs/latest/reference/proposals/proposals.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+| `accepted`    | The proposal has been accepted by the project maintainers.                        |
+| `rejected`    | The proposal has been rejected by the project maintainers.                        |
+| `implemented` | The proposal was accepted and has since been implemented.                         |
+
+## Proposal Lifecycle
+
+1. Author creates a ["Proposal request"](https://github.com/vmware-tanzu/kubeapps/issues/new?assignees=&labels=kind%2Fproposal&template=proposal-request.md&title=).
+2. Author adds the proposal by creating a PR in draft mode (authors can save their work until ready):
+   - this PR must include a markdown file in the root of the [proposals](./) folder, and
+   - this PR must reference the "Proposal request" (by adding the github ID).
```
Comment (antgamdia) [inline site/content/docs/latest/reference/proposals/proposals.md:44]:
```suggestion
   - this PR must reference the "Proposal request" (by adding the GitHub ID).
```

File: site/content/docs/latest/reference/proposals/proposals.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+3. When the author elaborates the proposal sufficiently and consider ready to be reviewed:
```
Comment (antgamdia) [inline site/content/docs/latest/reference/proposals/proposals.md:45]:
```suggestion
3. When the author elaborates the proposal sufficiently and considers ready to be reviewed:
```

File: site/content/docs/latest/reference/proposals/proposals.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+   - change the status of the "Proposal request" to `in-review`, and
+   - mark the PR as "Ready for Review".
+4. The community reviews the proposal by adding PR reviews in order to mature/converge on the proposal.
+5. When the maintainers reach consensus or supermajority to accept a proposal, they:
+   - change the status of the "Proposal request" to `accepted`,
+   - add the Engineering Decission Record (EDR) to the proposal (in the markdown file) including the following topics:
```
Comment (antgamdia) [inline site/content/docs/latest/reference/proposals/proposals.md:51]:
```suggestion
   - add the Engineering Decision Record (EDR) to the proposal (in the markdown file) including the following topics:
```

File: site/content/docs/latest/reference/proposals/proposals.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+     1. _considered options_,
+     2. _pros and cons_,
+     3. _decision drivers_, and
+     4. _decision outcome._
+   - merge the PR, thus adding the new proposal to the `main` branch,
+   - code implementation PRs are submitted separately to implement the solution.
+6. During implementation of an accepted proposal:
```
Comment (antgamdia) [inline site/content/docs/latest/reference/proposals/proposals.md:58]:
```suggestion
6. During the implementation of an accepted proposal:
```

File: site/content/docs/latest/reference/proposals/proposals.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+   - as each implementation PR is created, the "Proposal request" should be updated to link to the new implementation PR, and
+   - when all implementation PRs are merged, the "Proposal Request" should be updated to have status `implemented`, to list all the related PRs, and then, the "Proposal request" should be closed.
```
Comment (antgamdia) [inline site/content/docs/latest/reference/proposals/proposals.md:60]:
```suggestion
   - when all the implementation PRs are merged, the "Proposal Request" should be updated to the`implemented` status , to list all the related PRs, and then, the "Proposal request" should be closed.
```

File: site/content/docs/latest/reference/proposals/proposals.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+   - If it is discovered that significant unanticipated changes are needed to the proposal, then the implementation work should be paused and the proposal should be updated with the new details to be reviewed by the maintainers again before resuming implementation.
+7. When the maintainers do not reach consensus or supermajority, then the proposal is rejected, and they:
+   - may mark the status of the "Proposal request" to `rejected`,
```
Comment (antgamdia) [inline site/content/docs/latest/reference/proposals/proposals.md:63]:
```suggestion
   - may mark the status of the "Proposal request" as `rejected`,
```

File: site/content/docs/latest/reference/proposals/proposals.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+   - close the PR with a note explaining the rejection, and
+   - close the "Proposal request".
+8. Rejected proposal PRs (and the corresponding "Proposal request") may be reopened and moved back to `in-review` if there are material changes to the proposal which address the reasons for rejection.
+
+## Proposal Review
+
+Once a proposal PR marked as "Ready for Review", the community and all Kubeapps maintainers will review the proposal. The goal of the review is to gain an understanding of the problem being solved and the design of the proposed solution.
```
Comment (antgamdia) [inline site/content/docs/latest/reference/proposals/proposals.md:70]:
```suggestion
Once a proposal PR is marked as "Ready for Review", the community and all Kubeapps maintainers shall review the proposal. The goal of the review is to gain an understanding of the problem being solved and the design of the proposed solution.
```

File: site/content/docs/latest/reference/proposals/proposals.md [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+Maintainers will consider all aspects of the proposed problem and solution, including but not limited to:
+
+- Is the problem within scope for the project?
```
Comment (antgamdia) [inline site/content/docs/latest/reference/proposals/proposals.md:74]:
```suggestion
- Is the problem within the scope of the project?
```

File: site/content/docs/latest/reference/proposals/proposals.md [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+- Would the additional future cost of maintenance imposed by an implementation of the solution justify solving the problem?
+- Is the solution reasonably consistent with the rest of the project?
+- How does the solution impact the usability, security, scalability, performance, observability, and reliability of Kubeapps?
+- How might an implementation of the solution be architected and tested via automation?
+- What risks might be introduced by an implementation of the solution?
+- The opportunity cost of the time it would take to implement the solution, if the implementation is to be done by the maintainers.
+
+## Maintenance of Accepted Proposal Documents
+
+Proposal documents reflect a point-in-time design and decision. Once approved, they become historical documents, not living documents. There is no expectation that they will be maintained in the future. Instead, significant changes to a feature which came from a previous proposal should be proposed as a fresh proposal. New proposals should link to
```
Comment (antgamdia) [inline site/content/docs/latest/reference/proposals/proposals.md:84]:
```suggestion
Proposal documents reflect a point-in-time design and decision. Once approved, they become historical documents, not living documents. There is no expectation that they will be maintained in the future. Instead, significant changes to a feature that came from a previous proposal should be proposed as a fresh proposal. New proposals should link to
```

**Review by antgamdia — COMMENTED** [review]
Just minor copy issues. 
Also I noticed this PR is breaking the website preview? Can you run hugo locally to double-check it is just a preview error instead of changes breaking the website?

**Review by castelblanque — APPROVED** [review]

File: GOVERNANCE.md [hunk 11]
```diff
@@ -6,50 +6,27 @@ This document defines the project governance for Kubeapps.
 
 **Kubeapps**, an open-source project, is committed to building an open, inclusive, productive and self-governing open source community involved to simplifying how applications are deployed and managed in Kubernetes clusters. The community is governed by this document to define how the community should work together to achieve this goal.
```
Comment (castelblanque) [inline GOVERNANCE.md:7]:
```suggestion
**Kubeapps**, an open-source project, is committed to building an open, inclusive, productive and self-governing open source community devoted to simplifying how applications are deployed and managed in Kubernetes clusters. The community is governed by this document to define how the community should work together to achieve this goal.
```

Comment (castelblanque) [inline GOVERNANCE.md:7]:
```suggestion
**Kubeapps**, an open-source project, is committed to building an open, inclusive, productive and self-governing open source community focused on simplifying how applications are deployed and managed in Kubernetes clusters. The community is governed by this document to define how the community should work together to achieve this goal.
```

Comment (absoludity) [reply GOVERNANCE.md:7]: +1 for "focused on" rather than the other options.

Comment (absoludity) [reply GOVERNANCE.md:7]: Or "involved in" (only saw that afterwards), both are fine IMO.

**Review by absoludity — APPROVED** [review]
+1 with the feedback from different people here.

File: site/content/docs/latest/reference/proposals/proposals.md [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+   - change the status of the "Proposal request" to `accepted`,
+   - add the Engineering Decission Record (EDR) to the proposal (in the markdown file) including the following topics:
+     1. _considered options_,
+     2. _pros and cons_,
+     3. _decision drivers_, and
+     4. _decision outcome._
+   - merge the PR, thus adding the new proposal to the `main` branch,
+   - code implementation PRs are submitted separately to implement the solution.
+6. During implementation of an accepted proposal:
+   - as each implementation PR is created, the "Proposal request" should be updated to link to the new implementation PR, and
```
[the remark this answers is not in the document]
Comment (ppbaena) [reply site/content/docs/latest/reference/proposals/proposals.md:59]: +1 . 
I've added:
`The proposal file (.md file) should be updated by including the version where the proposal was released.`

**Comment by ppbaena** [comment]
> Also I noticed this PR is breaking the website preview? Can you run hugo locally to double-check it is just a preview error instead of changes breaking the website?

Done! Thank you Antonio I have updated a folder from design-proposals to proposals and it was not updated in the _index.md file for website. I've updated it and PR should pass all tests now.

**Pull request merged**